torch/csrc/Device.cpp

```diff
@@ -95,6 +96,24 @@ PyObject *THPDevice_index(THPDevice *self)
   END_HANDLE_TH_ERRORS
 }
 
+PyObject *THPDevice_hash(THPDevice *self)
+{
+  HANDLE_TH_ERRORS
+  switch (self->device.type()) {
+    case at::Device::Type::CPU: {
+      return THPUtils_packInt64(static_cast<int64_t>(self->device.index()));
+    }
+    case at::Device::Type::CUDA: {
+      if (self->device.index() > 254) {
+        AT_WARN("Device indices of > 254 might result in non-deterministic hashes");
```

Comment:
This warning is a little silly, because (1) it's not true as the code is written today, and (2) you wouldn't actually know what was meant here unless you read @apaszke's comment at https://github.com/pytorch/pytorch/issues/7228#issuecomment-386242514

Reply:
Yes, it does seem a bit uninformative and silly. I couldn't find out the source of non-determinism, so I thought I should keep it there, just in case.